### PR TITLE
add :raw datatype

### PR DESCRIPTION
**Pull request by psergus**
Add :raw datatype to implement JSON persistence to AWS DynamoDB

### CODE REVIEW

File: lib/dynamoid/persistence.rb [hunk 1]
```diff
@@ -134,6 +140,30 @@ def dynamo_type(type)
         end
       end
 
+      private
+
+            def transform_hash(hash)
```
Comment (twrodriguez) [inline lib/dynamoid/persistence.rb]:
```
                         Indentation?
```

File: lib/dynamoid/config.rb [hunk 2]
```diff
@@ -23,6 +23,7 @@ module Config
     option :use_ssl, :default => true
     option :port, :default => '443'
     option :identity_map, :default => false
+    option :convert_big_decimal, :default => false
```
Comment (twrodriguez) [inline lib/dynamoid/config.rb:26]: We may want to have this as a per-model or per-column option in the future, but this is good for now.

File: lib/dynamoid/persistence.rb (hunk 1 above)
Comment (twrodriguez) [inline lib/dynamoid/persistence.rb]: Also, `transform` doesn't describe which direction the translation is happening. I have to look at where it's called to understand that this is during the _deserializing_ process.

File: lib/dynamoid/persistence.rb [hunk 3]
```diff
@@ -134,6 +140,30 @@ def dynamo_type(type)
...
+        {}.tap do |h|
+          hash.each { |key, value| h[key.to_sym] = transform(value) }
+        end
+      end
+
+      def transform(val)
```
Comment (twrodriguez) [inline lib/dynamoid/persistence.rb]: ditto about non-descript `transform`

File: spec/dynamoid/persistence_spec.rb [hunk 4]
```diff
@@ -65,6 +65,12 @@
     expect(@user.send(:dump)[:name]).to eq 'Josh'
   end
 
+  it 'keeps raw Hash attributes as Hash' do
+    config = {acres: 5, trees: {cyprus: 30, poplar: 10, joshua: 1}, horses: ['Lucky', 'Dummy'], lake: 1, tennis_court: 1}
+    @addr = Address.new(:config => config)
+    expect(@addr.send(:dump)[:config]).to eq config
+  end
+
```
Comment (twrodriguez) [inline spec/dynamoid/persistence_spec.rb:91]: Given that we have an `if value.is_a? Hash` in `#undump_field`, could we add tests for the other data types and make sure they're sane? `Array`, `String`, `Numeric`, and `nil` would be great.

**Comment by twrodriguez** [comment]
:+1:
